ng-dev/release/notes/changelog.spec.ts

```diff
@@ -0,0 +1,83 @@
+import {existsSync, readFileSync} from 'fs';
+import {SemVer} from 'semver';
+import {dedent} from '../../utils/testing/dedent';
+import {getMockGitClient} from '../publish/test/test-utils/git-client-mock';
+import {Changelog, splitMarker} from './changelog';
+
+describe('Changelog', () => {
+  let changelog: Changelog;
+
+  beforeEach(() => {
+    const gitClient = getMockGitClient(
+      {owner: 'angular', name: 'dev-infra-test', mainBranchName: 'main'},
+      /* useSandboxGitClient */ false,
+    );
+    changelog = new Changelog(gitClient);
+  });
+
+  it('throws an error if it cannot find the anchor containing the version for an entry', () => {
+    expect(() => changelog.prependEntryToChangelog('does not have version <a> tag')).toThrow();
+  });
+
+  it('throws an error if it cannot determine the version for an entry', () => {
+    expect(() => changelog.prependEntryToChangelog(createChangelogEntry('NotSemVer'))).toThrow();
+  });
+
+  describe('adds entries to the changelog', () => {
+    it('creating a new changelog file if one does not exist.', () => {
+      expect(existsSync(changelog.changelogPath)).toBe(false);
+
+      changelog.prependEntryToChangelog(createChangelogEntry('0.0.0'));
+      expect(existsSync(changelog.changelogPath)).toBe(true);
+    });
+
+    it('not including a split marker when only one changelog entry is in the changelog.', () => {
```

Comment:
Just a small nit: I think jasmine tests should read as they appear in the code. e.g. `it should not include a split marker ..`. The current test description is not clear about the expected outcome.